lfs.h

```diff
@@ -84,6 +85,9 @@ enum lfs_error {
     LFS_ERR_NOMEM       = -12,  // No more memory available
     LFS_ERR_NOATTR      = -61,  // No data/attr available
     LFS_ERR_NAMETOOLONG = -36,  // File name too long
+#if LFS_THREAD_SAFE
+    LFS_ERR_LOCK        = -23,  // Failed to aquire lock
```

Comment:
Is this needed? littlefs's current strategy is to let any negative error codes go through (for example the block device callbacks). This lets the OS return more specific errors such as EBUSY or EDEADLK (looking at https://linux.die.net/man/3/pthread_mutex_lock).